# This file is part of HiGFXback

# requires
REQUIRES="autotools-wrappers-am-build libdrm-build libtsm-build udev-build"

PKG_CONFIG_PATH=/drm/share/pkgconfig pkg-config --exists --print-errors $REQUIRES || exit 1

# configure (ac-2.69; am-1.14; lt-2.4.2)
libtoolize -c -f; aclocal; autoconf; autoheader -f; automake -a -c
PKG_CONFIG_PATH=/drm/lib/pkgconfig LDFLAGS=-Wl,-rpath,/drm/lib ./configure --disable-static --with-video=drm2d --prefix=/drm

# build
make

# install
install -d $DESTDIR/drm/bin
install kmscon $DESTDIR/drm/bin
install -d $DESTDIR/drm/lib/kmscon
install .libs/mod-bbulk.so $DESTDIR/drm/lib/kmscon
install .libs/mod-unifont.so $DESTDIR/drm/lib/kmscon

# build.pc
install -d $DESTDIR/drm/share/pkgconfig
cat > $DESTDIR/drm/share/pkgconfig/kmscon-build.pc << EOF
Name: kmscon
Version: 20140708
Description: KMS/DRM terminal emulator
Requires: $REQUIRES

exec=\\
/drm/bin/kmscon \\
/drm/lib/kmscon/mod-bbulk.so \\
/drm/lib/kmscon/mod-unifont.so
EOF

exit
--- kmscon-20140708.orig/Makefile.am
+++ kmscon-20140708/Makefile.am
@@ -60,8 +60,7 @@
 	-fno-strict-aliasing \
 	-fvisibility=hidden \
 	-ffunction-sections \
-	-fdata-sections \
-	-fstack-protector
+	-fdata-sections
 AM_CPPFLAGS = \
 	-DBUILD_MODULE_DIR='"$(moduledir)"' \
 	-include $(top_builddir)/config.h \
--- kmscon-20140708.orig/configure.ac
+++ kmscon-20140708/configure.ac
@@ -55,7 +55,7 @@
 AC_SUBST(TSM_CFLAGS)
 AC_SUBST(TSM_LIBS)
 
-PKG_CHECK_MODULES([UDEV], [libudev >= 172])
+PKG_CHECK_MODULES([UDEV], [libudev >= 171])
 AC_SUBST(UDEV_CFLAGS)
 AC_SUBST(UDEV_LIBS)
 
--- kmscon-20140708.orig/src/text.c
+++ kmscon-20140708/src/text.c
@@ -438,7 +438,7 @@
 }
 
 int kmscon_text_draw_cb(struct tsm_screen *con,
-			uint32_t id, const uint32_t *ch, size_t len,
+			uint64_t id, const uint32_t *ch, size_t len,
 			unsigned int width,
 			unsigned int posx, unsigned int posy,
 			const struct tsm_screen_attr *attr,
--- kmscon-20140708.orig/src/text.h
+++ kmscon-20140708/src/text.h
@@ -101,7 +101,7 @@
 void kmscon_text_abort(struct kmscon_text *txt);
 
 int kmscon_text_draw_cb(struct tsm_screen *con,
-			uint32_t id, const uint32_t *ch, size_t len,
+			uint64_t id, const uint32_t *ch, size_t len,
 			unsigned int width,
 			unsigned int posx, unsigned int posy,
 			const struct tsm_screen_attr *attr,
